app/src/main/java/kr/co/marubook/presentation/chat/config/ChatListAdapter.kt

```diff
@@ -0,0 +1,90 @@
+package kr.co.marubook.presentation.chat.config
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import kr.co.marubook.databinding.ViewChatOtherBinding
+import kr.co.marubook.databinding.ViewChatUserBinding
+
+// 뭘 좋아할지 몰라 일단 어댑터를 2개 만들어 놓고 생각을 해보자
+val diffCallback = object : DiffUtil.ItemCallback<Message>() {
+    override fun areItemsTheSame(oldItem: Message, newItem: Message): Boolean {
+        when (oldItem.viewType) {
+            ChatListAdapter.RIGHT_CHAT -> {
+                if (newItem.viewType == ChatListAdapter.RIGHT_CHAT)
+                    return oldItem.messageIdx == newItem.messageIdx
+            }
+            ChatListAdapter.LEFT_CHAT -> {
+                if (newItem.viewType == ChatListAdapter.LEFT_CHAT)
+                    return oldItem.messageIdx == newItem.messageIdx
+            }
+        }
+        return false
+    }
+
+    override fun areContentsTheSame(oldItem: Message, newItem: Message): Boolean {
+        when (oldItem.viewType) {
+            ChatListAdapter.RIGHT_CHAT -> {
+                if (newItem.viewType == ChatListAdapter.RIGHT_CHAT)
+                    return oldItem.messageIdx == newItem.messageIdx
+            }
+            ChatListAdapter.LEFT_CHAT -> {
+                if (newItem.viewType == ChatListAdapter.LEFT_CHAT)
+                    return oldItem.messageIdx == newItem.messageIdx
+            }
+        }
+        return false
+    }
+}
```

Comment:
- 이 어댑터에서만 쓰일 디프이면 private val로 선언해주기!!
- 이건 상수로 간주될 수 있기 때문에(안 변경됨) DIFF_CALLBACK으로 해주는게 좀 더 좋음
- 이런경우에는 viewType이 똑같고 messageIdx가 같은 경우에만 비교해주고 나머지는 false 리턴해주는게 좀 더 깔끔하지 않을까? (아래에는 ``areItemsTheSame``만 예시듦)
```suggestion
val diffCallback = object : DiffUtil.ItemCallback<Message>() {
    override fun areItemsTheSame(oldItem: Message, newItem: Message) =
        oldItem.viewType == newItem.viewType && oldItem.messageIdx == newItem.messageIdx
}
```